KenticoCloud.Delivery.Tests/CodeFirstModelProviderTests.cs

```diff
@@ -0,0 +1,108 @@
+﻿using System;
+using System.Collections.Generic;
+using AngleSharp.Dom.Events;
+using FakeItEasy;
+using KenticoCloud.Delivery.InlineContentItems;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using Xunit;
+
+namespace KenticoCloud.Delivery.Tests
+{
+    public class CodeFirstModelProviderTests
+    {
+        [Fact]
+        // During processing of inline content items, item which detects circular dependency ( A refs B, B refs A ) should resolve resolved item
+        // as if there were no inline content items, which will prevent circular dependency
+        public void RetrievingContentModelWithCircularDependencyDoesNotCycle()
+        {
+            var fakeDeliverClient = A.Fake<IDeliveryClient>();
+            var codeFirstTypeProvider = A.Fake<ICodeFirstTypeProvider>();
+            A.CallTo(() => codeFirstTypeProvider.GetType(A<string>._)).Returns(typeof(ContentItemWithSingleRTE));
+            
+            var processor = new InlineContentItemsProcessor(null, null);
+            processor.RegisterTypeResolver(new RichTextInlineResolver());
+            var retriever = new CodeFirstModelProvider(fakeDeliverClient);
+            retriever.TypeProvider = codeFirstTypeProvider;
+            A.CallTo(() => fakeDeliverClient.InlineContentItemsProcessor).Returns(processor);
+
+            var item = JToken.FromObject(rt1);
+            var modularContent = JToken.FromObject(modularContentObject);
+
+            var result = retriever.GetContentItemModel<ContentItemWithSingleRTE>(item, modularContent);
+
+            Assert.Equal("<span>FirstRT</span><span>SecondRT</span><span>FirstRT</span>", result.RT);
```

Comment:
No, the first one in which circular dependency is processed as if there were no inline content items. This is first detected at rt1 where we find that we're processing object which is already being processed.